llama-index-server/README.md

```diff
@@ -32,24 +39,91 @@ def create_workflow() -> Workflow:
     )
 
 
-# Create an API server the workflow
+# Create an API server for the workflow
 app = LlamaIndexServer(
-    workflow_factory=create_workflow  # Supports Workflow or AgentWorkflow
+    workflow_factory=create_workflow,  # Supports Workflow or AgentWorkflow
+    env="dev",  # Enable development mode
+    include_ui=True,  # Include chat UI
+    starter_questions=["What can you do?", "How do I use this?"],
+    verbose=True
 )
 ```
 
-## Running the server
+## Running the Server
 
 - In the same directory as `main.py`, run the following command to start the server:
 
   ```bash
   fastapi dev
   ```
 
-- Making a request to the server
+- Making a request to the server:
 
   ```bash
   curl -X POST "http://localhost:8000/api/chat" -H "Content-Type: application/json" -d '{"message": "What is the weather in Tokyo?"}'
   ```
 
 - See the API documentation at `http://localhost:8000/docs`
+- Access the chat UI at `http://localhost:8000/` (Make sure you set the `env="dev"` or `include_ui=True` in the server configuration)
+
+## Configuration Options
+
+The LlamaIndexServer accepts the following configuration parameters:
+
+- `workflow_factory`: A callable that creates a workflow instance for each request
+- `logger`: Optional logger instance (defaults to uvicorn logger)
+- `use_default_routers`: Whether to include default routers (chat, static file serving)
+- `env`: Environment setting ('dev' enables CORS and UI by default)
+- `include_ui`: Whether to include the chat UI
+- `starter_questions`: List of starter questions for the chat UI
+- `verbose`: Enable verbose logging
+- `api_prefix`: API route prefix (default: "/api")
+- `ui_path`: Path for downloaded UI static files (default: ".ui")
+
+## Default Routers and Features
+
+### Chat Router
+
+The server includes a default chat router at `/api/chat` for handling chat interactions.
+
+### Static File Serving
```

Comment:
@leehuwuj as you see this doesn't make a lot of sense as our folder mapping is workflow specific.
Suggestion: we add a `file_server` parameter that we set to `['data', 'output']` for our example workflows. if it's empty there are no `api/files` routes
Do we still need then the `use_default_routers` parameter?